app/app.go

```diff
@@ -463,6 +481,52 @@ func NewApp(
 	return app
 }
 
+func inflationCalculator(ctx sdk.Context, minter minttypes.Minter, params minttypes.Params, bondedRatio sdk.Dec) sdk.Dec {
+	var deploymentParams deploymenttypes.Params
+	deploymentParamSubspace.GetParamSet(ctx, &deploymentParams)
+	tHalf := float64(deploymentParams.InflationDecayFactor)
+
+	// Number of hours in an year = 8766 (averaging the leap year hours)
+	// Number of minutes in an hour = 60
+	// Number of seconds in a minute = 60
+	// => 60 * 60 * 8766 = Number of seconds per year
+	t := ctx.BlockTime().Sub(genesisTime).Seconds() / (60 * 60 * 8766) // years passed
+	i := 100.0 * math.Pow(2, -t/tHalf)                                 // 100 = initial inflation (i0)
+	idealInflation := sdk.NewDec(int64(i))
+
+	// (1 - bondedRatio/GoalBonded) * InflationRateChange
+	inflationRateChangePerYear := sdk.OneDec().
+		Sub(bondedRatio.Quo(params.GoalBonded)).
+		Mul(params.InflationRateChange)
+	inflationRateChange := inflationRateChangePerYear.Quo(sdk.NewDec(int64(params.BlocksPerYear)))
+
+	// note inflationRateChange may be negative
+	currentInflation := minter.Inflation.Add(inflationRateChange)
+
+	// min, max currentInflation based on a defined range parameter 'r'
+	// currentInflation range = [I(t) - I(t) * R, I(t) + I(t) * R]
+	r, err := sdk.NewDecFromStr("0.05") // let's say we allow 5% variance
```

Comment:
Let's make the variance a parameter too.

Reply:
Done